products/experiments/backend/max_tools.py

```diff
@@ -0,0 +1,145 @@
+from typing import Any, Literal
+
+from pydantic import BaseModel, Field
+
+from posthog.models import Experiment, FeatureFlag
+from posthog.sync import database_sync_to_async
+
+from ee.hogai.tool import MaxTool
+
+
+class CreateExperimentArgs(BaseModel):
+    name: str = Field(description="Experiment name - should clearly describe what is being tested")
+    feature_flag_key: str = Field(
+        description="Feature flag key (letters, numbers, hyphens, underscores only). Will create a new flag if it doesn't exist."
+    )
+    description: str | None = Field(
+        default=None,
+        description="Detailed description of the experiment hypothesis, what changes are being tested, and expected outcomes",
+    )
+    type: Literal["product", "web"] = Field(
+        default="product",
+        description="Experiment type: 'product' for backend/API changes, 'web' for frontend UI changes",
+    )
+
+
+class CreateExperimentTool(MaxTool):
+    name: Literal["create_experiment"] = "create_experiment"
+    description: str = """
+Create a new A/B test experiment in the current project.
+
+Experiments allow you to test changes with a controlled rollout and measure their impact.
+
+Use this tool when the user wants to:
+- Create a new A/B test experiment
+- Set up a controlled experiment to test changes
+- Test variants of a feature with users
+
+Examples:
+- "Create an experiment to test the new checkout flow"
+- "Set up an A/B test for our pricing page redesign"
+- "Create an experiment called 'homepage-cta-test' to test different call-to-action buttons
+
+**IMPORTANT**: You must first find or create a multivariate feature flag using `create_feature_flag`, with at least two variants (control and test). Navigate to the feature flags page to create the flag, create the flag, then navigate back to the experiments page and use this tool to create the experiment."
+    """.strip()
+    context_prompt_template: str = "Creates a new A/B test experiment in the project"
+    args_schema: type[BaseModel] = CreateExperimentArgs
+
+    async def _arun_impl(
+        self,
+        name: str,
+        feature_flag_key: str,
+        description: str | None = None,
+        type: Literal["product", "web"] = "product",
+    ) -> tuple[str, dict[str, Any] | None]:
+        from posthog.exceptions_capture import capture_exception
+
+        # Validate inputs
+        if not name or not name.strip():
+            return "Experiment name cannot be empty", {"error": "invalid_name"}
+
+        if not feature_flag_key or not feature_flag_key.strip():
+            return "Feature flag key cannot be empty", {"error": "invalid_flag_key"}
+
+        @database_sync_to_async
+        def create_experiment() -> Experiment:
+            # Check if experiment with this name already exists
+            existing_experiment = Experiment.objects.filter(team=self._team, name=name, deleted=False).first()
+            if existing_experiment:
```

Comment:
Question: I feel confused, but `Experiment` doesn't have any indexes? It's not a "fix something", but rather a surprise - am I missing it?